tests/page/matchers.misc.spec.ts

```diff
@@ -23,3 +24,23 @@ it('should outlive frame navigation', async ({ page, server }) => {
   }, 1000);
   await expect(page.locator('.box').first()).toBeEmpty();
 });
+
+it('should print no-locator-resolved error when locator matcher did not resolve to any element', async ({ page, server }) => {
+  const myLocator = page.locator('.nonexisting');
+  const expectWithShortLivingTimeout = expect.configure({ timeout: 10 });
+  const locatorMatchers = [
+    () => expectWithShortLivingTimeout(myLocator).toBeAttached(),                 // Boolean matcher
+    () => expectWithShortLivingTimeout(myLocator).toHaveJSProperty('abc', 'abc'), // Equal matcher
+    () => expectWithShortLivingTimeout(myLocator).not.toHaveText('abc'),          // Text matcher - pass / string
+    () => expectWithShortLivingTimeout(myLocator).not.toHaveText(/abc/),          // Text matcher - pass / RegExp
+    () => expectWithShortLivingTimeout(myLocator).toContainText('abc'),           // Text matcher - fail
+  ];
+  for (const matcher of locatorMatchers) {
+    await it.step(matcher.toString(), async () => {
+      const error = await matcher().catch(e => e);
+      expect(error).toBeInstanceOf(Error);
+      expect(error.message).toContain(`waiting for locator('.nonexisting')`);
+      expect(stripAnsi(error.message)).toMatch(/Received:  ?"?<element\(s\) not found>/);
```

Comment:
Now that we only have 6 cases, maybe inlining them with exact expected error message would be easier to understand?

Reply:
Not a huge fan, since it would make the test much larger. This underlines our motivation that the `waiting for locator` and `Received` value should be displayed in all the scenarios.